File: SECRET_PROJECT_WEB/src/types/Channel/Channel.ts
import type { ChannelMessage } from "../ChannelMessage/ChannelMessage";
import type { ChannelUser } from "../ChannelUser/ChannelUser";

export type Channel = {
  users: ChannelUser[];
  messages: ChannelMessage[];
};

export type ChannelDto = {
  id: string;
  name: string;
  channelAvatarUrl: string;
  createdAt: string;
};

export type AddChannelRequest = {
  name: string;
  channelAvatarUrl: string;
  adminId: string;
};

export type JoinChannelRequest = {
  userId: string;
  channelId: string;
};

File: SECRET_PROJECT_SERVER/Secret_Project_Backend/SecretProject.Platform.Data/DataStore/Entities/Channel.cs
using SecretProject.Platform.Data.DataStore.Models;

namespace SecretProject.Platform.Data.DataStore.Entities
{
    public class Channel
    {
        public required Guid Id { get; set; }
        public required string Name { get; set; } = string.Empty;
        public string? ChannelAvatarUrl { get; set; }
        public required DateTime CreatedAt { get; set; }
        public Guid? AdminId { get; set; }
        //public ApplicationUser? Admin { get; set; }
        public Guid? RoomId { get; set; }
        public Room? Room { get; set; }
        public virtual ICollection<ChannelMessage>? ChannelMessages { get; set; }
        public virtual ICollection<ChannelUser>? ChannelUsers { get; set; }
    }
}
